src/groundlight/internalapi.py

```diff
@@ -225,3 +235,125 @@ def _get_detector_by_name(self, name: str) -> Detector:
                 f"We found multiple ({parsed['count']}) detectors with the same name. This shouldn't happen.",
             )
         return Detector.parse_obj(parsed["results"][0])
+
+    @RequestsRetryDecorator()
+    def submit_image_query_with_inspection(self, detector_id: str, image: ByteStreamWrapper, inspection_id: str) -> str:
+        """Submits an image query to the API and returns the ID of the image query.
+        The image query will be associated to the inspection_id provided.
+        """
+        url = f"{self.configuration.host}/posichecks?inspection_id={inspection_id}&predictor_id={detector_id}"
+
+        headers = self._headers()
+        headers["Content-Type"] = "image/jpeg"
+
+        response = requests.request("POST", url, headers=headers, data=image.read())
+
+        if not is_ok(response.status_code):
+            logger.info(response)
+            raise InternalApiError(
+                status=response.status_code,
+                reason=f"Error submitting image query with inspection ID {inspection_id} on detector {detector_id}",
+                http_resp=response,
+            )
+
+        return response.json()["id"]
+
+    @RequestsRetryDecorator()
+    def start_inspection(self) -> str:
+        """Starts an inspection, returns the ID."""
+        url = f"{self.configuration.host}/inspections"
+
+        headers = self._headers()
+
+        response = requests.request("POST", url, headers=headers, json={})
+
+        if not is_ok(response.status_code):
+            raise InspectionError(f"Error starting inspection. Status code: {response.status_code}")
+
+        return response.json()["id"]
+
+    @RequestsRetryDecorator()
+    def update_inspection_metadata(self, inspection_id: str, user_provided_key, user_provided_value) -> None:
```

Comment:
I think the key should be a string.

```suggestion
    def update_inspection_metadata(self, inspection_id: str, user_provided_key: str, user_provided_value) -> None:
```

Reply:
Yes, the are always strings. I will add the type hints.